### PR TITLE
yarpfixmanager RTF plugin fix

### PR DESCRIPTION
**Pull request by aerydna**
yarpfixmanager search the fixture file with the absolute path if the relative path failed

### CODE REVIEW

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 1]
```diff
@@ -50,8 +51,14 @@ bool YarpFixManager::setup(int argc, char** argv) {
 
         fixtureName = rf.find("fixture").asString();
         std::string appfile = rf.findFileByName(std::string("fixtures/"+fixtureName).c_str());
+        if (!appfile.size())
+        {
+            yInfo("trying to load from absolute path");
```
Comment (traversaro) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: This print is a bit out of context without more information, for example the class from which it is emitted.

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 2]
```diff
@@ -50,8 +51,14 @@ bool YarpFixManager::setup(int argc, char** argv) {
 
         fixtureName = rf.find("fixture").asString();
         std::string appfile = rf.findFileByName(std::string("fixtures/"+fixtureName).c_str());
+        if (!appfile.size())
```
Comment (Nicogene) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: use `appfile.empty()`

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 3]
```diff
@@ -51,13 +51,13 @@ bool YarpFixManager::setup(int argc, char** argv) {
 
         fixtureName = rf.find("fixture").asString();
         std::string appfile = rf.findFileByName(std::string("fixtures/"+fixtureName).c_str());
-        if (!appfile.size())
+        if (!appfile.empty())
```
Comment (pattacini) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: @aerydna 
the condition `!appfile.size()` should be `appfile.empty()` instead

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 4]
```diff
@@ -51,13 +51,13 @@ bool YarpFixManager::setup(int argc, char** argv) {
...
         {
             yInfo("yarpmanager: trying to load fixture file from absolute path");
             appfile = rf.findFileByName(std::string(fixtureName).c_str());
         }
 
-        RTF_ASSERT_ERROR_IF(appfile.size(),
+        RTF_ASSERT_ERROR_IF(appfile.empty(),
```
Comment (pattacini) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: same comment applies here

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 5]
```diff
@@ -50,8 +51,14 @@ bool YarpFixManager::setup(int argc, char** argv) {
...
         std::string appfile = rf.findFileByName(std::string("fixtures/"+fixtureName).c_str());
-        RTF_ASSERT_ERROR_IF(appfile.size(),
-                            RTF::Asserter::format("yarpmanager cannot find apllication file %s. Is it in the 'fixtures' folder?",
+        if (appfile.empty())
+        {
+            yInfo("yarpmanager: trying to load fixture file from absolute path");
+            appfile = rf.findFileByName(std::string(fixtureName).c_str());
+        }
+
+        RTF_ASSERT_ERROR_IF(!appfile.empty(),
```
Comment (aerydna) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: mmm... due to the rtf issue 67 of robot testing repository ("RTF_TEST_FAIL_IF macro logic is inverted") this line is correct but pretty confusing

Comment (pattacini) [reply src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: Go on, just wanted to remark that `size()>0` has an inverted logic wrt `empty()` method 😉

**Comment by drdanz** [comment]
Merged, thanks!

**Pull request merged**